torchao/dtypes/nf4tensor.py

```diff
@@ -585,5 +593,21 @@ def linear_nf4(input: torch.Tensor, weight: NF4Tensor) -> torch.Tensor:
 
 
 def to_nf4(tensor, block_size: int = 64, scaler_block_size: int = 256):
-    tensor1 = tensor.to(torch.bfloat16)
-    return NF4Tensor.from_tensor(tensor1, block_size, scaler_block_size)
+    return NF4Tensor.from_tensor(tensor, block_size, scaler_block_size)
+
+
+NF4_TORCH_FUNCTIONS = {}
+
+
+def implements_torch_function(torch_function):
+    def decorator(func):
+        functools.update_wrapper(func, torch_function)
+        NF4_TORCH_FUNCTIONS[torch_function] = func
+        return func
+
+    return decorator
+
+
+@implements_torch_function(torch.Tensor.to)
+def function_to_dtype(*args, **kwargs):
+    return args[0].get_original_weight().to(args[1])
```

Comment:
nit there is a few ways you can call to that this isn't robust too I would imagine

Reply:
do you mean 3 ways to call `.to` ?
* `to(dtype, non_blocking=False, copy=False, memory_format=torch.preserve_format) `
* `torch.to(device=None, dtype=None, non_blocking=False, copy=False, memory_format=torch.preserve_format)`
* `torch.to(other, non_blocking=False, copy=False) `
I can raise unimplemented if it helps

Reply:
yeah, I think in theory they all can be supported its just that the other args/kwargs are getting dropped as implemented

Reply:
got you. will see if I can pass in args/kwargs instead of dropping them

Reply:
updated the PR to include args and kwargs instead of dropping. passing to dispatch if not implemented